python/revng/pypeline/task/savepoint.py

```diff
@@ -84,31 +84,28 @@ def run(
             set(incoming.keys()) | set(outgoing.keys())
         ), "SavePoint containers must be a subset of incoming and outgoing requests."
 
-        # Cache the containers present for this configuration
         for decl in self.to_save:
-            if decl not in incoming:
-                continue
-            if len(incoming[decl]) == 0:
-                continue
-            storage_provider.put(
-                ContainerLocation(
+            # Cache the containers present for this configuration
```

Comment:
I'm not entirely sure what's going on here.

Can you expand on the commit message?